File: Data Structures & Algorithms/sort-colors/submission-0.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        vector<int> counts(3);

        for (int& num : nums) {
            counts[num] += 1;
        }

        int index = 0;
        for (int i = 0; i < counts.size(); ++i) {
            while (counts[i]-- > 0) {
                nums[index++] = i;
            }
        }
        
    }
};

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        def recurse(index):
            if index == len(nums):
                return [[]]

            result = []
            perms = recurse(index + 1)
            for p in perms:
                for j in range(len(p)+1):
                    pCopy = p.copy()
                    pCopy.insert(j, nums[index])
                    result.append(pCopy)

            return result
        return recurse(0)
            

        